### ISSUE TITLE
GraalVM/SubstrateVM's native-image build breaks after update from Micronaut 2.0.2 to 2.0.3

### ISSUE THREAD

**Issue by csh0711** [micronaut-projects/micronaut-core#4218]
My [basic Micronaut example](https://github.com/csh0711/micronaut-data-graalvm-kotlin) (using Kotlin and Micronaut Data) demonstrates how to create a native image with GraalVM.
This works fine with Micronaut 2.0.2. But after updating to Micronaut 2.0.3 I get the following GraalVM error when executing ` native-image` :

```
Fatal error:java.lang.IllegalArgumentException: java.lang.IllegalArgumentException: Class com.typesafe.config.Config cannot be instantiated reflectively. It must be a non-abstract instance type.
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:600)
        at java.base/java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1006)
        at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:480)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:349)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:508)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:114)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:537)
Caused by: java.lang.IllegalArgumentException: Class com.typesafe.config.Config cannot be instantiated reflectively. It must be a non-abstract instance type.
        at org.graalvm.sdk/org.graalvm.nativeimage.hosted.RuntimeReflection.registerForReflectiveInstantiation(RuntimeReflection.java:120)
        at io.micronaut.context.env.hocon.nativeimage.HoconFeature.registerClassForRuntimeReflection(HoconFeature.kt:44)
        at io.micronaut.context.env.hocon.nativeimage.HoconFeature.beforeAnalysis(HoconFeature.kt:37)
        at com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$7(NativeImageGenerator.java:693)
        at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:70)
        at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:693)
        at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:555)
        at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:468)
        at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Error: Image build request failed with exit status 1
```

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Clone https://github.com/csh0711/micronaut-data-graalvm-kotlin

2. Run the following steps with branch `master` (contains _Micronaut 2.0.2_ app):
```
sdk use java 20.2.0.r11-grl
./gradlew clean assemble
native-image --no-server -cp build/libs/footballmanager-0.1-all.jar
```
And voilà - the native image (`footballmanager`) is created

3. Run the previous steps with branch `2_0_3_native_image_error`  (contains _Micronaut 2.0.3_ app):
You'll get the error mentioned above when running `native-image --no-server -cp build/libs/footballmanager-0.1-all.jar`

### Expected Behaviour

The native image should be build using Micronaut 2.0.3 and GraalVM CE 20.2.0

### Actual Behaviour
`native-image` breaks

### Environment Information

- **Operating System**: macOS Mojave
- **Micronaut Version:** 2.0.3
- **JDK Version:** GraalVM CE 20.2.0.

### Example Application

https://github.com/csh0711/micronaut-data-graalvm-kotlin
-> Branch `master` is working (Micronaut 2.0.2), branch `2_0_3_native_image_error` is erroneous  (Micronaut 2.0.3)

**Comment by csh0711** [comment]
@graemerocher Sorry to bother, just one question: means _closed_ "will be fixed with 2.1.0" or "not reproducible"?
Thanks a lot! :)

**Comment by graemerocher** [comment]
Fixed in 2.1.0

**Comment by csh0711** [comment]
Just a short confirmation:  Works like a charm with 2.1.0 :-)
